include/crow/http_response.h

```diff
@@ -246,13 +246,17 @@ namespace crow
                 code = 200;
                 this->add_header("Content-length", std::to_string(file_info.statbuf.st_size));
 
-                if (extension != "")
+                if (!extension.empty())
```

Comment:
If extension is empty then `Content-length` will be set and `Content-Type` not? Also wouldn't it be better to check if extension exists in map to avoid handling exceptions?

Reply:
Re `Content-Type` - not sending that header when the file has no extension does match previous behavior (see [here](https://github.com/CrowCpp/Crow/blob/db53bd619b51d4e45c03206adcb38eaf9683562a/include/crow/http_response.h#L249)), but now that I hear you say it that does sound wrong.  Looking at the [HTTP spec](https://datatracker.ietf.org/doc/html/rfc2616#section-7.2.1), it appears that, when a client receives a response missing `Content-Type`, it should introspect the body and guess the content type, or assume `application/octet-stream`.  I suppose we could make that assumption ourselves and send `application/octet-stream` when there's no extension, but this would deny clients that introspection opportunity.  What do you think?

I'll change it to use `find()` instead of exception handling if that's what you prefer.

Reply:
I guess you're right and we should leave it as it is ;)